tests/tests.py

```diff
@@ -0,0 +1,97 @@
+import json
+from unittest import expectedFailure
+
+from django.core.urlresolvers import reverse, resolve
+from django.test import SimpleTestCase, RequestFactory
+
+from edtf.views import edtf_form, boolean_result, result_json
+
+
+class TestURLs(SimpleTestCase):
+
+    def test_edtf_form_url(self):
+        view = resolve(reverse('edtf:edtf_form')).func
+        self.assertEqual(view, edtf_form)
+
+    def test_results_json_url(self):
+        view = resolve(reverse('edtf:results_json')).func
+        self.assertEqual(view, result_json)
+
+
+class TestEdtfFormView(SimpleTestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
+        request = self.factory.get('/')
+        response = edtf_form(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_correct_template_used(self):
+        url = reverse('edtf:edtf_form')
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, 'edtf/edtf_form.html')
+
+
+class TestBooleanResultView(SimpleTestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
```

Comment:
boolean_result() implies that if the `date` value is an empty string it should give a 400. Because the code is written `date is ''` instead of `date == ''` making a request with date='' is giving a 200 when it should probably be 400. In that case, I would say the expected status_code for this test should be 400 and that this should be expected to fail because of a bug in the view code. That being said, boolean_result does not seem to be used, seems a bit broken, and we could probably skip testing it, remove the view code and the commented out urls.py stuff related to it (and while we are at it, remove the urlerror() function in views.py that is also apparently not used).